certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -225,6 +238,32 @@ def updated_mod_ssl_conf_digest(self):
         """Full absolute path to digest of updated SSL configuration file."""
         return os.path.join(self.config.config_dir, constants.UPDATED_MOD_SSL_CONF_DIGEST)
 
+    def openssl_version(self):
+        """Lazily retrieve openssl version"""
+        if self._openssl_version:
+            return self._openssl_version
+        # Step 1. Check for LoadModule directive
+        try:
+            ssl_module_location = self.parser.modules['ssl_module']
+        except KeyError:
+            return None
+        if not ssl_module_location:
+            return None
```

Comment:
Did you consider logging a warning in this case too?

Reply:
If I did, I can't remember why I decided not to; adding warnings now.